tests/phpunit/TestFramework/Coverage/LineRangeCalculatorTest.php

```diff
@@ -0,0 +1,175 @@
+
+namespace App\Config;
+
+use Monolog\Logger;
+
+class ProdConfig extends AbstractConfig
+{
+    /**
+     * @return array<string, mixed>
+     */
+    public function getConfig(): array
+    {
+        $cacheDir = $this->getCacheDir();
+        $logDir = $this->getLogDir();
+
+        return [// line 19 of code snippet
+            'cors' => [
+                'allow-origin' => [],
+                'allow-methods' => ['DELETE', 'GET', 'POST', 'PUT'],
+                'allow-headers' => [
+                    'Accept',
+                    'Content-Type',
+                ],
+                'allow-credentials' => false,
+                'expose-headers' => [],
+                'max-age' => $findMe,
+            ],
+            'debug' => false,
+            'doctrine.dbal.db.options' => [
+                'configuration' => [
+                    'cache.result' => ['type' => 'apcu'],
+                ],
+                'connection' => [
+                    'driver' => 'pdo_pgsql',
+                    'charset' => 'utf8',
+                    'user' => getenv('DATABASE_USER'),
+                    'password' => getenv('DATABASE_PASS'),
+                    'host' => getenv('DATABASE_HOST'),
+                    'port' => getenv('DATABASE_PORT'),
+                    'dbname' => getenv('DATABASE_NAME'),
+                ],
+            ],
+            'doctrine.orm.em.options' => [
+                'cache.hydration' => ['type' => 'apcu'],
+                'cache.metadata' => ['type' => 'apcu'],
+                'cache.query' => ['type' => 'apcu'],
+                'proxies.dir' => $cacheDir.'/doctrine/proxies',
+            ],
+            'monolog' => [
+                'name' => 'petstore',
+                'path' => $logDir.'/application.log',
+                'level' => Logger::NOTICE,
+            ],
+            'routerCacheFile' => $cacheDir.'/routes.php',
+        ]; // line 58 of code snippet
+    }
+
+    public function getEnv(): string
+    {
+        return 'prod';
+    }
+}
+
+PHP;
+
+        $nodes = $parser->parse($code);
+        $traverser = new NodeTraverser();
+        $spy = $this->spyTraverser();
+        $traverser->addVisitor(new ParentConnectorVisitor());
+        $traverser->addVisitor($spy);
+        $traverser->traverse($nodes);
+        $range = $spy->range;
+        $this->assertSame(range(19, 58), $range);
+    }
+
+    public function test_it_calculates_range_outside_of_array(): void
+    {
+        $parser = Container::create()->getParser();
+        $code = <<<'PHP'
+<?php
+
+function foo(): void
+{
+    (static function() {
+        $a = $findMe;
+    })();
+}
+
+PHP;
+
+        $nodes = $parser->parse($code);
+        $traverser = new NodeTraverser();
+        $spy = $this->spyTraverser();
+        $traverser->addVisitor(new ParentConnectorVisitor());
+        $traverser->addVisitor($spy);
+        $traverser->traverse($nodes);
+        $range = $spy->range;
+        $this->assertSame([6], $range);
```

Comment:
Might be just me but I find that adding some spaces here would add quite a bit of readability:

```php
        $traverser = new NodeTraverser();
        $traverser->addVisitor(new ParentConnectorVisitor());
        $traverser->addVisitor($spy = $this->spyTraverser());

        $traverser->traverse($parser->parse($code));

        $range = $spy->range;

        $this->assertSame([6], $range);
```